src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -246,18 +268,33 @@ object RuleUtils {
           }
         }
 
+        val readSchema = if (filesDeleted.isEmpty) {
```

Comment:
`newSchema`? Also, can you add a comment why you have a different branch for delete (e.g., need to utilized DATA_FILE_NAME_COLUMN, etc.)? BTW, can you just append this data file name column to the `StructType(index.schema.filter(baseRelation.schema.contains(_)))` or no?

Reply:
I moved the condition in filter:
```
val newSchema = StructType(index.schema.filter(s =>
            baseRelation.schema.contains(s) || (filesDeleted.nonEmpty && s.name.equals(
              IndexConstants.DATA_FILE_NAME_COLUMN))))
```